crates/oxc_transformer/src/plugins/styled_components.rs

```diff
@@ -680,13 +683,15 @@ impl<'a> StyledComponents<'a, '_> {
         })
     }
 
-    /// Returns the display name which infers the component name or gets from the file name.
-    fn get_display_name(&self, ctx: &TraverseCtx<'a>) -> Option<&'a str> {
-        let component_name = Self::get_component_name(ctx);
+    /// Returns the block name based on the file stem or parent directory name.
+    fn get_block_name(&mut self, ctx: &TraverseCtx<'a>) -> Option<&'a str> {
```

Comment:
Block name is never inserted into AST. I think it should be a `String`, rather than allocating it into arena.

Reply:
Done!